### PR TITLE
Add support for regional NEG stripping of maxUtilization based on url

### PR DESCRIPTION
**Pull request by slevenick**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/7051

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: Added support to `google_compute_backend_service` for setting a serverless regional network endpoint group as `backend.group`
```

### CODE REVIEW

File: templates/terraform/encoders/backend_service.go.erb [hunk 1]
```diff
@@ -43,8 +43,12 @@ for _, backendRaw := range backends {
 	if !ok {
 		continue
 	}
-	if strings.Contains(backendGroup.(string), "global/networkEndpointGroups") {
-		// Remove `max_utilization` from any backend that belongs to a global NEG. This field
+	match, err := regexp.MatchString("projects/(?P<project>[^/]+)/regions/(?P<region>[^/]+)/networkEndpointGroups/(?P<name>[^/]+)", backendGroup.(string))
+	if err != nil {
+		return nil, err
+	}
+	if match || strings.Contains(backendGroup.(string), "global/networkEndpointGroups") {
```
Comment (danawillow) [inline templates/terraform/encoders/backend_service.go.erb:50]: It feels a bit extra to me to have both regex matching and strings.contains. does it make sense to do just one or the other?

File: third_party/terraform/tests/resource_compute_backend_service_test.go.erb [hunk 2]
```diff
@@ -1508,3 +1532,47 @@ resource "google_compute_http_health_check" "zero" {
 }
 `, serviceName, sampleRate, checkName)
 }
+
+<% unless version == 'ga' -%>
+func testAccComputeBackendService_regionNegBackend(suffix string) string {
+	return fmt.Sprintf(`
+resource "google_compute_backend_service" "backend" {
+  name                            = "backend%s"
```
Comment (danawillow) [inline third_party/terraform/tests/resource_compute_backend_service_test.go.erb:1540]: this should be prefixed with tf-test so it gets swept (probably the other resources too)

File: templates/terraform/encoders/backend_service.go.erb (hunk 1 above)
Comment (slevenick) [reply templates/terraform/encoders/backend_service.go.erb:50]: Yeah, string.Contains worked for global NEGs as the key string `global/networkEndpointGroups` is contiguous, needed a regex to look for `regions/{{region}}/networkEndpointGroups`. I'll make them both use a regex

File: templates/terraform/encoders/backend_service.go.erb [hunk 3]
```diff
@@ -43,10 +43,14 @@ for _, backendRaw := range backends {
 	if !ok {
 		continue
 	}
-	if strings.Contains(backendGroup.(string), "global/networkEndpointGroups") {
-		// Remove `max_utilization` from any backend that belongs to a global NEG. This field
-		// has a default value and causes API validation errors
-		backend["maxUtilization"] = nil
+	serverlessNegMatchers := [2]*regexp.Regexp{regexp.MustCompile("projects/(?P<project>[^/]+)/regions/(?P<region>[^/]+)/networkEndpointGroups/(?P<name>[^/]+)"), regexp.MustCompile("projects/(?P<project>[^/]+)/global/networkEndpointGroups/(?P<name>[^/]+)")}
```
Comment (danawillow) [inline templates/terraform/encoders/backend_service.go.erb:46]: regexes only need to match part of the string, so I think just using `(?:global|region/[^/]+)/networkEndpointGroups` as the regex would work too and be a bit less text. any reason to do this way instead?

**Pull request merged**